stingray/varenergyspectrum.py

```diff
@@ -116,72 +171,107 @@ class VarEnergySpectrum(object):
+
+    @property
+    def energy(self):
+        return np.sum(self.energy_intervals, axis=1) / 2
+
+    def _analyze_inputs(self):
+        events1 = self.events1
+        events2 = self.events2
+        common_gti = events1.gti
+        if events2 is None or events2 is events1:
+            self.events2 = self.events1
+            self.same_events = True
+        else:
+            common_gti = cross_two_gtis(events1.gti, events2.gti)
+            self.same_events = False
+        self.gti = common_gti
 
-    def _decide_ref_intervals(self, channel_band, ref_band):
-        """
-        Ensures that the ``channel_band`` (i.e. the band of interest) is
-        not contained within the ``ref_band`` (i.e. the reference band)
+    def _create_empty_spectrum(self):
 
-        Parameters
-        ----------
-        channel_band : iterable of type ``[elow, ehigh]``
-            The lower/upper limits of the energies to be contained in the band
-            of interest
+        if self.return_complex:
+            dtype = complex
+        else:
+            dtype = float
 
-        ref_band : iterable
-            The lower/upper limits of the energies in the reference band
+        self.spectrum = np.zeros(len(self.energy_intervals), dtype=dtype) + np.nan
+        self.spectrum_error = np.zeros_like(self.spectrum, dtype=dtype) + np.nan
 
-        Returns
-        -------
-        ref_intervals : iterable
-            The channels that are both in the reference band in not in the
-            bands of interest
-        """
-        channel_band = np.asarray(channel_band)
-        ref_band = np.asarray(ref_band)
-        if len(ref_band.shape) <= 1:
-            ref_band = np.asarray([ref_band])
-        if check_separate(ref_band, [channel_band]):
-            return np.asarray(ref_band)
-        not_channel_band = [[0, channel_band[0]],
-                            [channel_band[1], np.max([np.max(ref_band),
-                                                      channel_band[1] + 1])]]
-
-        return cross_two_gtis(ref_band, not_channel_band)
-
-    def _construct_lightcurves(self, channel_band, tstart=None, tstop=None,
-                               exclude=True, only_base=False):
+    def _get_times_from_energy_range(self, events, erange, use_pi=False):
+        if use_pi:
+            energies = events.pi
+        else:
+            energies = events.energy
+        mask = (energies >= erange[0]) & (energies < erange[1])
+        return events.time[mask]
+
+    def _get_good_frequency_bins(self, freq=None):
+        if freq is None:
+            n_bin = np.rint(self.segment_size / self.bin_time)
+            freq = fftfreq(int(n_bin), self.bin_time)
+            freq = freq[freq > 0]
+        good = (freq >= self.freq_interval[0]) & (freq < self.freq_interval[1])
+        return good
+
+    def _get_ctrate(self, events):
+        return get_average_ctrate(events, self.gti, self.segment_size)
+
+    def _decide_ref_intervals(self, *args):
```

Comment:
Same here

Reply:
We can discuss if we want to remove them altogether, but I think that they have some usefulness